src/mainloop.rkt

```diff
@@ -326,15 +329,20 @@
 (define (run-improve! initial specification context pcontext)
   (timeline-event! 'preprocess)
   (define-values (simplified preprocessing)
-    (find-preprocessing initial specification context))
+    (find-preprocessing initial specification context)) ;; Eventually we can remove the preprocessing return here, but that requires fixing 
+                                                        ;; calls here & the sandbox call & subsequent usage that still require it
   (timeline-push! 'symmetry (map ~a preprocessing))
   (define pcontext* (preprocess-pcontext context pcontext preprocessing))
-  (match-define (and alternatives (cons (alt best _ _) _))
+  (match-define (and alternatives (cons (alt best _ _ _) _))
     (mutate! simplified context pcontext* (*num-iterations*)))
+    
   (timeline-event! 'preprocess)
-  (define preprocessing*
-    (remove-unnecessary-preprocessing best context pcontext preprocessing))
-  (values alternatives preprocessing*))
+  (define final-alts
+    (for/list ([altern alternatives])
+      (make-alt-preprocessing (alt-expr altern) (remove-unnecessary-preprocessing best context pcontext (alt-preprocessing altern)))))
```

Comment:
Wait wait wait I think this totally throws away the existing history?! You probably want `alt-add-preprocessing` or something.